### PR TITLE
ci: automate upgrade AWS CDK

### PR DESCRIPTION
**Pull request by WinterYukky**
### Reason for this change
The AWS CDK version for this repository has not been updated. However, since it is inefficient for contributors to update this every time, we need automate process.

### Description of changes
I implemented GitHub Actions that create a pull request to upgrade the AWS CDK version at 0 UTC every day.

### Description of how you validated changes
I tested on https://github.com/WinterYukky/projen-auto-upgrade-cdk/pull/2.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/open-constructs/aws-cdk-library/blob/main/CONTRIBUTING.md)

---

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license_

### CODE REVIEW

File: .projenrc.ts [hunk 1]
```diff
@@ -64,4 +65,52 @@ project.addTask('integ:update', {
...
+    pullRequests: JobPermission.WRITE,
+  },
+  runsOn: ['ubuntu-latest'],
+  steps: [
+    {
+      uses: 'actions/checkout@v4',
+    },
+    {
+      name: 'Install dependencies',
+      run: 'yarn install',
```
Comment (go-to-k) [inline .projenrc.ts:85]: Thanks for this PR! Should we use npm?

see: https://github.com/open-constructs/aws-cdk-library/issues/9

**Comment by mrgrain** [comment]
I don't think this is a good idea. It unnecessarily forces users to update to the latest version of aws-cdk-lib when they just want to upgrade `@open-constructs/aws-cdk`.

But then I also think that changing a peerDeps constraint should require a major version change. 😅 

As a possible solution I can only suggest what I have suggested before: A well-define regular cycle for the release of new major versions which will include lifting the minimal version requirement to whatever is latest at that time.

File: .projenrc.ts (hunk 1 above)
Comment (WinterYukky) [reply .projenrc.ts]: Oops, thanks!

**Comment by WinterYukky** [comment]
Ah, I see. That's right.
First, We should documentation about major version release schedule.

**Pull request closed without merging**